src/main/java/ca/nrc/cadc/beacon/web/config/VOSpaceServiceConfigMgr.java

```diff
@@ -0,0 +1,164 @@
+    public static final String VOSPACE_DEFAULT_SERVICE_NAME_KEY = "org.opencadc.vosui.service.default";
+
+    // Used to construct service-specific property keys (ie KEY_BASE + {svc name} + {rest of key}
+    public static final String KEY_BASE = "org.opencadc.vosui.";
+    public static final String NODE_URI_KEY = ".node.resourceid";
+    public static final String USER_HOME_KEY = ".user.home";
+    public static final String SERVICE_PROPER_NAME_KEY = ".service.proper.name";
+    public static final String SERVICE_TYPE_KEY = ".service.type";
+    public static final String SERVICE_RESOURCEID_KEY = ".service.resourceid";
+
+    public VOSpaceServiceConfigMgr(ApplicationConfiguration appConfig) {
+        this.serviceConfigMap = new TreeMap<>();
+        this.loadConfig(appConfig);
+    }
+
+    public void loadConfig(ApplicationConfiguration appConfig) {
+        if (serviceConfigMap.size() != 0) {
+            this.serviceConfigMap = new TreeMap<>();
+            this.serviceList = new ArrayList<>();
+        }
+
+        try {
+            this.defaultServiceName = appConfig.lookup(VOSPACE_DEFAULT_SERVICE_NAME_KEY);
+
+            // Get all VOSpace services named in the config file
+            this.serviceList = Arrays.asList(appConfig.lookupAll(VOSPACE_SERVICE_NAME_KEY));
+
+            for (String storageServiceName: this.serviceList) {
+
+                log.debug("adding vospace service to map: " + VOSPACE_SERVICE_NAME_KEY + ": " + serviceList.toString());
+
+                // Values that will vary by configuration
+                String properName = appConfig.lookup(KEY_BASE + storageServiceName + SERVICE_PROPER_NAME_KEY);
+                String serviceType = appConfig.lookup(KEY_BASE + storageServiceName + SERVICE_TYPE_KEY);
+                URI vospaceResourceID = appConfig.lookupServiceURI(KEY_BASE + storageServiceName + SERVICE_RESOURCEID_KEY, new URI(""));
+                URI nodeResourceID = appConfig.lookupServiceURI(KEY_BASE + storageServiceName + NODE_URI_KEY, new URI(""));
+
+                VOSpaceServiceConfig newConfig = new VOSpaceServiceConfig(storageServiceName,
+                    properName, serviceType, vospaceResourceID, nodeResourceID);
+
+                String userHomeDir = KEY_BASE + storageServiceName + USER_HOME_KEY;
+                log.debug("user home directory: " + userHomeDir);
+                String userHomeValue = appConfig.lookup(userHomeDir);
+                if (StringUtil.hasLength(userHomeDir)) {
+                    newConfig.homeDir = userHomeValue;
+                }
+
+                log.debug("node resource id base: " + nodeResourceID);
+
+                // Set to map
+                this.serviceConfigMap.put(storageServiceName, newConfig);
+
+            } // end for (String sotrageServiceName: this.serviceList) { ...
+
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Unable to init context. Required configuration missing.", e);
```

Comment:
It would be beneficial here to list the missing configuration.  Also, it's not _just_ an exception that will cause this; the URIs on lines 124 and 125 could be empty, and therefore missing, but no exception will be thrown.

Reply:
Added more explicit messages for missing configuration.